x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/KnnFunctionIT.java

```diff
@@ -30,42 +34,69 @@
 
 import static org.elasticsearch.index.IndexMode.LOOKUP;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.xpack.esql.DenseVectorFieldTypeIT.ALL_DENSE_VECTOR_INDEX_TYPES;
 import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
 
 public class KnnFunctionIT extends AbstractEsqlIntegTestCase {
 
-    private final Map<Integer, List<Float>> indexedVectors = new HashMap<>();
+    private final Map<Integer, List<Number>> indexedVectors = new HashMap<>();
     private int numDocs;
     private int numDims;
 
+    private final DenseVectorFieldMapper.ElementType elementType;
+    private final String indexType;
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() throws Exception {
```

Comment:
You could simplify this a bit, with something like this (untested): 

```
public static Iterable<Object[]> parameters() throws Exception {
      return ALL_DENSE_VECTOR_INDEX_TYPES.stream()
          .filter(indexType -> indexType.contains("flat") == false)
          .map(indexType -> new Object[] { DenseVectorFieldMapper.ElementType.FLOAT, indexType })
          .collect(Collectors.toList());
  }
```

Reply:
I find good ol' loops better in terms of clarity here. Also, soon enough we'll need to add the bytes as well.